oak_attestation_verification/src/verifier.rs

```diff
@@ -720,6 +714,82 @@ fn verify_measurement_digest(
     }
 }
 
+/// Adapted version of verify_measure_digest() for kernels.
+fn verify_kernel_measurement_digest(
+    image_measurement: &RawDigest,
+    setup_data_measurement: &RawDigest,
+    now_utc_millis: i64,
+    endorsement: Option<&TransparentReleaseEndorsement>,
+    reference_value: &KernelBinaryReferenceValue,
+) -> anyhow::Result<()> {
+    let actual_image = raw_to_hex_digest(image_measurement);
+    let actual_setup_data = raw_to_hex_digest(setup_data_measurement);
+    match reference_value.r#type.as_ref() {
+        Some(kernel_binary_reference_value::Type::Skip(_)) => Ok(()),
+        Some(kernel_binary_reference_value::Type::Endorsement(public_keys)) => {
+            let endorsement =
+                endorsement.context("matching endorsement not found for reference value")?;
+            verify_binary_endorsement(
+                now_utc_millis,
+                &endorsement.endorsement,
+                &endorsement.rekor_log_entry,
+                &public_keys.endorser_public_key,
+                &public_keys.rekor_public_key,
+            )?;
+            // Parse endorsement statement and verify attachment digest.
```

Comment:
This can probably be encapsulated as a function since the same will be done every time there is an attachment.

Reply:
I agree. Let's put this off until the stage0 case is addressed in an upcoming PR.